src/main/kotlin/org/rust/lang/core/psi/RustCompositeElement.kt

```diff
@@ -10,3 +13,21 @@ interface RustCompositeElement : PsiElement {
 
 val RustCompositeElement.containingMod: RustMod?
     get() = PsiTreeUtil.getStubOrPsiParentOfType(this, RustMod::class.java)
+
+/**
+ * Returns the level of the given lint for this element.
+ */
+fun PsiElement.lintLevel(lint: RustLint): RustLintLevel {
+    var level: RustLintLevel? = null
+    if (this is RustDocAndAttributeOwner) {
```

Comment:
You can write this as `ancestors.filterIsInstance<RustDocAndAttributeOwner>` I think.